### PR TITLE
Latest terraform

### PR DESCRIPTION
**Pull request by josmo**
@jmccann I haven't looked at all the details around this yet but it's what I needed to get it working with 0.9.4 version of terraform.

### CODE REVIEW

File: plugin.go [hunk 1]
```diff
@@ -102,7 +103,7 @@ func installCaCert(cacert string) *exec.Cmd {
 
 func exportSecrets(secrets map[string]string) {
 	for k, v := range secrets {
-		os.Setenv(fmt.Sprintf("%s", k), fmt.Sprintf("%s", os.Getenv(v)))
+             os.Setenv(k, v)
```
Comment (jmccann) [inline plugin.go]: I think removing the `fmt.Srptinf` is OK, but seems we accidentally got rid of the `os.Getenv(v)` part as well?  I think we want to keep that.

File: plugin.go [hunk 2]
```diff
@@ -58,6 +58,7 @@ func (p Plugin) Exec() error {
 		commands = append(commands, remoteConfigCommand(remote))
 	}
 	commands = append(commands, getModules())
+        commands = append(commands, initCommand())
```
Comment (jmccann) [inline plugin.go]: Can we fix spacing please?

File: plugin.go [hunk 3]
```diff
@@ -146,8 +147,14 @@ func validateCommand() *exec.Cmd {
...
 
+func initCommand() * exec.Cmd {
+   return exec.Command("terraform", "init",)
+}
+
+
 func planCommand(variables map[string]string, secrets map[string]string, parallelism int, targets []string) *exec.Cmd {
-	args := []string{
+	exportSecrets(secrets)
+        args := []string{
```
Comment (jmccann) [inline plugin.go]: Fix spacing please

File: plugin.go [hunk 4]
```diff
@@ -146,8 +147,14 @@ func validateCommand() *exec.Cmd {
...
 }
 
+func initCommand() * exec.Cmd {
+   return exec.Command("terraform", "init",)
+}
+
+
 func planCommand(variables map[string]string, secrets map[string]string, parallelism int, targets []string) *exec.Cmd {
-	args := []string{
+	exportSecrets(secrets)
```
Comment (jmccann) [inline plugin.go]: `exportSecrets` is already called earlier @ https://github.com/josmo/drone-terraform/blob/1847409c8d1cb007712b217181f1031e788beb28/plugin.go#L49

File: plugin.go [hunk 5]
```diff
@@ -158,10 +165,6 @@ func planCommand(variables map[string]string, secrets map[string]string, paralle
 		args = append(args, "-var")
 		args = append(args, fmt.Sprintf("%s=%s", k, v))
 	}
-	for k, v := range secrets {
```
Comment (jmccann) [inline plugin.go]: I don't think we want to remove ability to pass secrets as vars?

File: plugin.go (hunk 1 above)
Comment (jmccann) [inline plugin.go]: Also, not sure if we need to fix spacing here as well?

**Comment by jmccann** [comment]
@josmo Thanks for this PR.  I think I'm going to favor #41 over this though as it doesn't have any updates required so far to move forward.  Thanks again though!

**Comment by josmo** [comment]
@jmccann #44  should actually fix the mess I was trying to do above :) . I didn't quite get how the plugin worked, but #44 should have the correct fix for .6 and up

**Pull request closed without merging**